src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBuffer.cs

```diff
@@ -103,32 +136,42 @@ private void AppendValue(ViewBufferValue value)
         private ViewBufferPage GetCurrentPage()
         {
             ViewBufferPage page;
-            if (Pages.Count == 0)
```

Comment:
`_currentPage` has to be updated in `MoveTo()` destination too. But it works if I update the property directly in `AddPage()`.